kiosk/src/Components/AddingGame.tsx

```diff
@@ -0,0 +1,161 @@
+import { useEffect, useState } from "react";
+import { Kiosk } from "../Models/Kiosk";
+import { KioskState } from "../Models/KioskState";
+import configData from "../config.json"
+import "../Kiosk.css";
+import AddGameButton from "./AddGameButton";
+import {QRCodeSVG} from 'qrcode.react';
+interface IProps {
+    kiosk: Kiosk
+  }
+
+const AddingGame: React.FC<IProps> = ({ kiosk }) => {
+    // TODO: update the urls to be more flexible for production code.
+    const [kioskCode, setKioskCode] = useState("");
+    const [renderQRCode, setRenderQRCode] = useState(true);
+    const [menuButtonSelected, setMenuButtonState] = useState(false);
+    const [qrCodeButtonSelected, setQrButtonState] = useState(false);
+    const kioskCodeUrl = "http://localhost:3000/static/kiosk/";
+
+    const updateLoop = () => {
+        if (!menuButtonSelected && kiosk.gamepadManager.isDownPressed()) {
+            setMenuButtonState(true);
+            if (qrCodeButtonSelected) {
+                setQrButtonState(false);
+            }
+        }
+        if (menuButtonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+            kiosk.showMainMenu();
+        }
+        if (!renderQRCode && kiosk.gamepadManager.isUpPressed()) {
+            setMenuButtonState(false);
+            setQrButtonState(true);
+        }
+        if (qrCodeButtonSelected && kiosk.gamepadManager.isAButtonPressed()) {
+            setRenderQRCode(true);
+        }
+    }
+
+    useEffect(() => {
+        let intervalId: any = null;
+        intervalId = setInterval(() => {
+            updateLoop();
+        }, configData.GamepadPollLoopMilli);
+        
+        return () => {
+            if (intervalId) {
+                clearInterval(intervalId);
+            }
+        };
+
+    });
+
+    useEffect(() => {
+        async function generateKioskCode() {
+            const codeGenerationUrl = "https://staging.pxt.io/api/kiosk/newcode"
+            const response = await fetch(codeGenerationUrl);
+            if (!response.ok) {
+                throw new Error("Unable to generate kiosk code");
+            }
+
+            try {
+                const newKioskCode = (await response.json()).code; 
+                setKioskCode(newKioskCode);
+                await addGameToKiosk(newKioskCode);
+            }
+            catch (error) {
+                setRenderQRCode(false);
+                throw new Error("Unable to generate kiosk code");
+            }
+        }
+
+        generateKioskCode();
+
+        function addGameToKiosk(kioskCode: string) {
+            const timeoutDuration = 600000; // wait for 10 minutes until the kiosk code expires
+            const whenToPull = 5000; // wait for five seconds to poll for game code data
```

Comment:
Is this supposed to be whenToPoll?

Reply:
Yes haha, I thought I changed it but must have forgotten to.

Reply:
Hmm.. Weird, I have all of the changes locally and it says that I am up to date with the remote, but I see the older versions of the files... I wonder what went wrong.